### PR TITLE
chore: clean up renderer logic

### PR DESCRIPTION
**Pull request by j-luong**

### CODE REVIEW

File: internal/presenters/presenter_local_finding.go [hunk 1]
```diff
@@ -27,27 +27,16 @@ var embeddedFiles embed.FS
...
+	Input        *local_models.LocalFinding
+	config       configuration.Configuration
+	writer       io.Writer
+	templateImpl map[string]TemplateImplFunction
 }
 
 type FilteredFindings struct {
-	OpenFindings    []*PresentationFindingResource
-	IgnoredFindings []*PresentationFindingResource
+	OpenFindings    []local_models.FindingResource
```
Comment (PeterSchafer) [inline internal/presenters/presenter_local_finding.go:38]: Question: should we just have all findings and methods injected to filter by state?

File: internal/presenters/presenter_local_finding.go [hunk 2]
```diff
@@ -171,68 +151,39 @@ func (p *LocalFindingPresenter) RenderTemplate(templateFiles []string, mimeType
...
+	filteredFindings := filterFindingsByIgnored(p.Input.Findings, shouldShowIgnored)
 
 	err = mainTmpl.Execute(p.writer, struct {
 		Summary         SummaryData `json:"summary"`
-		OpenFindings    []*PresentationFindingResource
-		IgnoredFindings []*PresentationFindingResource
-		ShowIgnored     bool
-		ShowDivider     bool
-		SeverityFilter  string
+		OpenFindings    []local_models.FindingResource
```
Comment (PeterSchafer) [inline internal/presenters/presenter_local_finding.go:165]: Question: Do we still need this local struct?

File: internal/presenters/presenter_local_finding.go [hunk 3]
```diff
@@ -171,69 +146,40 @@ func (p *LocalFindingPresenter) RenderTemplate(templateFiles []string, mimeType
...
-		}
-
-		return 0
-	})
-
-	return result
-}
-
-func filterOutIgnoredFindings(findings []local_models.FindingResource, showIgnored bool, sortOrder []string) (filtered FilteredFindings) {
+func filterFindingsByIgnored(findings []local_models.FindingResource, showIgnored bool) (openFindings []local_models.FindingResource, ignoredFindings []local_models.FindingResource) {
```
Comment (thisislawatts) [inline internal/presenters/presenter_local_finding.go:173]: **issue:** refactor to make this generic

File: internal/presenters/templates/local_finding.tmpl [hunk 4]
```diff
@@ -1,15 +1,15 @@
 {{- define "details" }}
 {{- if gt ($.OpenFindings | len) 0 }}{{ "Open Issues" | title }}
```
Comment (thisislawatts) [inline internal/presenters/templates/local_finding.tmpl:2]: **issue:** Remove usage of injected `OpenFindings` and `IgnoredFindings`, instead process LocalFindings struct directly.

File: internal/presenters/templates/local_finding.tmpl [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
         {{- end }}
 {{- end }}
-{{- if .ShowDivider}}{{divider}}{{- end}}
-{{- if $.ShowIgnored }}{{"Ignored Issues" | title }}
+{{- if or (and (gt ($.OpenFindings | len) 0) (gt ($.IgnoredFindings | len) 0)) (eq (getValueFromConfig "include-ignores") "true") }}{{divider}}{{- end}}
+{{- if eq (getValueFromConfig "include-ignores") "true" }}{{"Ignored Issues" | title }}
 
 {{- if gt ($.IgnoredFindings | len) 0 }}
-        {{- range $finding := $.IgnoredFindings}}
+        {{- range $finding := ($.IgnoredFindings | sortFindingBySevAsc)}}
```
Comment (thisislawatts) [inline internal/presenters/templates/local_finding.tmpl:12]: **issue:** `sortFindingBySevAsc` remove use of hard coded value. Severity sort order is provided as part of the `Summary` object

Comment (thisislawatts) [reply internal/presenters/templates/local_finding.tmpl]: Based on https://github.com/snyk/go-application-framework/pull/269/commits/ad55021a94f37c68bc5a5469389c0c4bd64fa96e

File: internal/presenters/presenter_local_finding.go (hunk 3 above)
Comment (thisislawatts) [reply internal/presenters/presenter_local_finding.go]: Currently tied to FindingResource type. Given a FindingsList is a primitive in our domain this seems reasonable for the time being.

**Review by sandor-trombitas — APPROVED** [review]
lgtm 👍

**Pull request merged**